internal/impl/nats/processor_kv.go

```diff
@@ -275,7 +300,7 @@ func (p *kvProcessor) Process(c context.Context, msg *service.Message) (service.
 		p.addMetadata(m, key, 0, nats.KeyValuePurge)
 		return service.MessageBatch{m}, nil
 
-	case kvHistory:
+	case kvpOperationHistory:
 		entries, err := kv.History(key)
```

Comment:
I just noticed that `kv.History()` supports wildcards in the `key`. Not sure if this is by design @codegangsta, but I think it would be better to adjust the docs and pass in `keys_filter` instead of `key` for this operation as well. Also, see the comment about changing this one to return an array of objects instead of a batch of messages.

Reply:
We should absolutely make it consistent, I added some questions about the use of `keys_filter` in another comment to discuss.